week03/249. Group Shifted Strings Prem.cpp

```diff
@@ -0,0 +1,52 @@
+//tc: O(n*m), mc: O(n)
+//n = strings length, m=word length
+
+class Solution {
+public:
+
+    static const int BASE = 997;
+    static const long long MOD = 101103107109113LL;
+
+    long long getHash(string &s)
+    {
+        long long str_hash = 1;
+        for( auto ch : s ){
+
+
+            str_hash = BASE*str_hash + ((ch - s[0] + 26) % 26);

```

Comment:
Why don't you create a separate function to calculate the hash value? It is not clear what are you doing here in gibberish form.

Reply:
I made this hash function thinking as a separate function for calculating hash value against "each string" from group strings. That's why , I made a function to return hash value against each string and passed the hash value as a key to an unordered_map.